test/encoding.ts

```diff
@@ -26,6 +26,16 @@ export interface Trade {
   owner: string;
 }
 
+export class Interaction {
```

Comment:
ubernit: Personally, I think we should probably avoid a class in this case when it can be a simple interface (like the other `Trade` and `Order` types for example). This allows simple objects to be used as interactions instead of requiring them to have a specific `prototype` (which is what classes do in JS/TS). Kind of how it was with the other structs.

The other upside is that it allows you to define methods that do the encoding and ensure the parameters show up in the write order. For example:
```ts
new Interaction(address, callData);
new Interaction(callData, address);
```

The compiler accepts both happily, and at a glance there is no way to tell which one is correct without inspecting the `Interaction` source (or hovering over it if you are in an IDE that supports this, not so obvious when looking at a diff in Github for example).

You chould define something like:
```ts
interface Interaction {
  target: string,
  callData: string,
}
function encodeInteraction(trade: Interaction): AbiInteraction { ... }
```

This way the following calls are both correct, and the order the properties are specified doesn't matter. 

```ts
encodeInteraction({
  target: address,
  callData,
})
encodeInteraction({
  callData,
  target: address,
})
```

Also, in general JS/TS makes working with these plain interface types easier in general IMO.